src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -600,6 +601,9 @@ private void InitializeCustomNodeManager()
                     {
                         customNodeSearchRegistry.Remove(info.FunctionId);
                         SearchModel.Remove(searchElement);
+                        var workspacesToRemove = _workspaces.FindAll(w => w is CustomNodeWorkspaceModel
+                            && (w as CustomNodeWorkspaceModel).CustomNodeId == id);
+                        workspacesToRemove.ForEach(w => RemoveWorkspace(w));
                     }
```

Comment:
I can't fault that, except being curious if we could have used `_workspaces.RemoveAll` instead.